node/core/dispute-coordinator/src/scraping/mod.rs

```diff
@@ -201,12 +203,13 @@ impl ChainScraper {
 		sender: &mut Sender,
 		block_number: BlockNumber,
 		block_hash: Hash,
-	) -> Result<()>
+	) -> Result<Vec<CandidateEvent>>
 	where
 		Sender: overseer::DisputeCoordinatorSenderTrait,
 	{
+		let events = get_candidate_events(sender, block_hash).await?;
 		// Get included and backed events:
-		for ev in get_candidate_events(sender, block_hash).await? {
+		for ev in &events {
```

Comment:
I wonder if it is a good idea to return only the included events here.

If I'm not mistaken - we no longer need the backed ones and we do filter out the included events on another place in the code (`prioritize_newly_included`). We can save one filtering there. 

But this is just a thought -  it's okay to leave it as it is now.

Reply:
Oh wait... we need both for clearing spam filters.

Reply:
You caught it, but that's right! We still clear spam for both.